Fix webapp configs

jimallman opened the pull request:
Cleanup and consolidation of Ansible configuration of our web-apps

- simplify API base URLs and endpoints
- add explicit production API base URLs (used in dev deployment for some API calls)
- rename some confusing Ansible variables (and filenames) and web-app config keys
- check current playbook's roles (via `role_names`) to adjust apache config for API vs. web-app vs. others TBD

NB - This is working on `devtree`, along with [a companion`fix-webapp-configs` PR in opentree](https://github.com/OpenTreeOfLife/opentree/pull/1312).

bredelings committed:
Replace all otcetera branches with 'development'

snacktavish committed:
Merge pull request #28 from OpenTreeOfLife/unify-otc-branches1

Replace all otcetera branches with 'development'

jimallman committed:
Planning and organization of domain URLs

jimallman committed:
Clarify comment

jimallman committed:
Use new config files; rename app URL (vs endpoint)

jimallman committed:
Consolidate API base URLs ("domains") for webapp config

snacktavish committed:
clean up variables to split prod and dev

snacktavish committed:
frogot one file

jimallman committed:
Use abbreviated API base URLs everywhere

jimallman committed:
Config-file substitions to match curation config

jimallman committed:
Rename web-app URL for consistency

jimallman committed:
Remove unused endpoint vars

snacktavish committed:
test prod dpeloy

snacktavish committed:
phylesystamapi template

jimallman committed:
Strip unwanted quotes from CACHED base URLs

jimallman committed:
Standardize on new API base URL names

jimallman committed:
Use a dynamic path to repos (to fix taxo-browser)

jimallman committed:
Use the general shared apache config by default

jimallman committed:
Add conditional test for phylesystemapi config

jimallman committed:
More debug info on role_names

jimallman commented on roles/apache/tasks/main.yml:
```diff
@@ -52,10 +52,9 @@
 - name: write /etc/apache2/opentree-shared.conf
   template:
-    src: templates/opentree_devapi_conf.j2
+    src: "{{ 'templates/opentree_devapi_conf.j2' if 'phylesystemapi' in role_names else 'templates/opentree_shared_conf.j2' }}" 
```
@snacktavish This test of `... in role_names` seems to work well (see also debug output in subsequent tasks).

snacktavish committed:
oti url

jimallman committed:
More debugging of role_name vars

jimallman committed:
Ignore goofy role-name vars, use custom var

It works consistently and serves as a good flag for this purpose.

jimallman committed:
Point to devapi for most API calls (devtree)

snacktavish committed:
use github path for directory name

jimallman committed:
Add base URls to phylesystemapi config

This should allow us to safely fetch from API methods on another server,
without risking caching results from random other domains.

snacktavish committed:
fix docstore paths

snacktavish committed:
Merge branch 'fix-webapp-configs' into phylesystem-api-pyramid

snacktavish committed:
Merge pull request #34 from OpenTreeOfLife/phylesystem-api-pyramid

Phylesystem api pyramid

snacktavish committed:
update vars for devphylesystemapi

jimallman commented:
This could probably use a final review for our always-on-production endpoints. If necessary, we can use both dev (default) and production versions of a method, by defining two different endpoints [in our config](https://github.com/OpenTreeOfLife/ot-ansible/pull/32/files#diff-bb366cb4fc45fe2453381a0f70634d53947c2f7b4e61b8f3f79f980ade98d5f6), like so (and using both names as appropriate in the web-app code):
```inputrc
# a typical endpoint, on both default/dev and production
getContextForNames_url = {default_apis}/v3/tnrs/infer_context
PRODUCTION_getContextForNames_url = {production_apis}/v3/tnrs/infer_context

# and this is a cached call for both default/dev and production
getSynthesisSourceList_url = {CACHED_default_apis}/v3/tree_of_life/about
PRODUCTION_getSynthesisSourceList_url = {CACHED_production_apis}/v3/tree_of_life/about
```

jimallman committed:
Provide dev API stubs (third-level domain names)

Add default stub values that point to production; all dev
server groups should specify the dev equivalents.

jimallman committed:
Revert mod-wsgi to python2 (for web2py)

I believe modwsgi is only used for our web2py web apps, so it should
be the python2 version; otherwise we get immediate errors in web2py even
when pointing to the proper pytyhon2.7 venv

jimallman committed:
typo

snacktavish commented:
This branch was merged into development - can we delete this branch "fix-webapp-configs" and work from development?

jimallman committed:
Migrate webhook creation from old germinator

This should(!) create webhooks to nudge the study and amendment indices
after pushing changes to Github.

NB - Still using old callback URLs here.

jimallman committed:
Move var to shared location

snacktavish commented:
I merged most of these into development, and used that to deploy production phylesystem-api!
Can we move the webhook deploy commit to a new branch off of development?

jimallman committed:
Install a webhook via GitHub API (still testing)

@snacktavish, this should work, but apparently my local OPENTREEAPI_OAUTH_TOKEN
is outdated or no longer valid. Also, we might want to update the webhook URLs
to call directly to otindex (as discussed). Or abandon this, if we want
to punt and keep adding these webhooks manually.

jimallman committed:
Disable webhook tasks for now

jimallman commented:
> Can we move the webhook deploy commit to a new branch off of development?

Agreed. Moving these changes to a new branch, and closing this one.
